src/inmanta/moduletool.py

```diff
@@ -584,42 +603,54 @@ def list(self, requires: bool = False) -> None:
         project = Project.get()
         project.get_complete_ast()
 
-        names = sorted(project.modules.keys())
-        specs = project.collect_imported_requirements()
+        names: Sequence[str] = sorted(project.modules.keys())
+        specs: Dict[str, List[InmantaModuleRequirement]] = project.collect_imported_requirements()
         for name in names:
 
             name_length = max(len(name), name_length)
-            mod = Project.get().modules[name]
+            mod: Module = Project.get().modules[name]
             version = str(mod.version)
             if name not in specs:
                 specs[name] = []
 
-            try:
-                if project.install_mode == InstallMode.master:
-                    reqv = "master"
-                else:
-                    release_only = project.install_mode == InstallMode.release
-                    versions = ModuleV1.get_suitable_version_for(name, specs[name], mod._path, release_only=release_only)
-                    if versions is None:
-                        reqv = "None"
+            generation: str = str(mod.GENERATION.name)
+
+            reqv: str
+            matches: bool
+            editable: bool
+            if isinstance(mod, ModuleV1):
+                try:
+                    if project.install_mode == InstallMode.master:
+                        reqv = "master"
                     else:
-                        reqv = str(versions)
-            except Exception:
-                LOGGER.exception("Problem getting version for module %s" % name)
-                reqv = "ERROR"
+                        release_only = project.install_mode == InstallMode.release
+                        versions = ModuleV1.get_suitable_version_for(name, specs[name], mod._path, release_only=release_only)
+                        if versions is None:
+                            reqv = "None"
+                        else:
+                            reqv = str(versions)
+                except Exception:
+                    LOGGER.exception("Problem getting version for module %s" % name)
+                    reqv = "ERROR"
+                matches = version == reqv
+                editable = True
+            else:
+                reqv = ",".join(req.version_spec_str() for req in specs[name] if req.specs) or "*"
+                matches = all(version in req for req in specs[name])
+                editable = mod.is_editable()
 
             version_length = max(len(version), len(reqv), version_length)
 
-            table.append((name, version, reqv, version == reqv))
+            table.append((name, generation, editable, version, reqv, matches))
```

Comment:
The `version_length` variable is not used anywhere. We can clean that up in my opinion.

Reply:
Done.